### PR TITLE
feat: allow for relative imports from packages

### LINKED ISSUE #877: Allow locators within a package to be relatively-referenced
### Background & motivation

Currently, even if I want to import a Starlark file right next to my `main.star`, I have to type out the full name of the package like so:
```
adjacent = import_module("github.com/org/package/adjacent.star")
```
This quickly gets cumbersome.

### Desired behaviour

I'd expect to have a shorthand for specifying the full thing like Javascript allows: https://blog.logrocket.com/understanding-relative-absolute-imports-next-js/#:~:text=Next.-,js%20relative%20imports,.%2F%20%2C%20and%20so%20on.&text=Usually%2C%20when%20we%20have%20files%20like%20Header.

AFAICT, if we keep track of the path of the current Starlark file that's executing, we should be able to support a relative import syntax like:

```
adjacent = import_module("./adjacent.star")
```

Technically, this would simply be an alternate syntax for the full path - the shorthand would simply get translated into the long-form under the covers. Of course, if something doesn't exist, we'd still catch it at validation time.

LIMITATION: users would still not be able to do relative imports beyond the package boundary! (this is intentional)

### How important is this to you?

Painful; the lack of this feature makes using Kurtosis frictionful.

### PR DESCRIPTION
**Pull request by h4ck3rk3y**
## Description:
Allow for relative imports from packages

## Is this change user facing?
YES

### CODE REVIEW

File: core/server/api_container/server/api_container_service.go [hunk 1]
```diff
@@ -138,6 +138,10 @@ func (apicService ApiContainerService) RunStarlarkPackage(args *kurtosis_core_rp
 	relativePathToMainFile := args.GetRelativePathToMainFile()
 	mainFuncName := args.GetMainFunctionName()
 
+	if relativePathToMainFile == "" {
```
Comment (leoporoli) [inline core/server/api_container/server/api_container_service.go:141]: nice!

File: core/server/api_container/server/startosis_engine/startosis_packages/git_package_content_provider/parsed_git_url.go [hunk 2]
```diff
@@ -103,6 +103,11 @@ func parseGitURL(packageURL string) (*ParsedGitURL, *startosis_errors.Interpreta
 	return parsedGitURL, nil
 }
 
+func (parsedUrl *ParsedGitURL) getLocatorRelativeToThisURL(relativeUrl string) string {
```
Comment (leoporoli) [inline core/server/api_container/server/startosis_engine/startosis_packages/git_package_content_provider/parsed_git_url.go]: the method name says `relative` and the returned variable name says `absolute` is this ok?

File: core/server/api_container/server/startosis_engine/startosis_packages/mock_package_content_provider/mock_package_content_provider.go [hunk 3]
```diff
@@ -62,6 +64,14 @@ func (provider *MockPackageContentProvider) GetModuleContents(fileInsidePackageU
 	return string(fileContent), nil
 }
 
+func (provider *MockPackageContentProvider) GetAbsoluteLocatorForRelativeModuleLocator(_ string, relativeOrAbsoluteModulePath string) (string, *startosis_errors.InterpretationError) {
+	if strings.HasPrefix(relativeOrAbsoluteModulePath, startosis_constants.GithubDomainPrefix) {
+		return relativeOrAbsoluteModulePath, nil
+	}
+	// implement properly
```
Comment (leoporoli) [inline core/server/api_container/server/startosis_engine/startosis_packages/mock_package_content_provider/mock_package_content_provider.go]: is this pending? or can we remove the comment? if it's pending probably we could replace it with a TODO

File: internal_testsuites/starlark/valid-package-with-relative-imports/main.star [hunk 4]
```diff
@@ -0,0 +1,6 @@
+lib = import_module("./src/lib.star")
+password = read_file("./static_files/password.txt")
+
+def run(plan):
+    plan.print(lib.NAME)
+    plan.print(password)
```
Comment (leoporoli) [inline internal_testsuites/starlark/valid-package-with-relative-imports/main.star]: final line missing

File: internal_testsuites/starlark/valid-package-with-relative-imports/src/lib.star [hunk 5]
```diff
@@ -0,0 +1 @@
+NAME = "John Doe"
```
Comment (leoporoli) [inline internal_testsuites/starlark/valid-package-with-relative-imports/src/lib.star]: final line missing

File: internal_testsuites/starlark/valid-package-with-relative-imports/static_files/password.txt [hunk 6]
```diff
@@ -0,0 +1 @@
+Open Sesame
```
Comment (leoporoli) [inline internal_testsuites/starlark/valid-package-with-relative-imports/static_files/password.txt:1]: :)

**Review by leoporoli — APPROVED** [review]

File: core/server/api_container/server/startosis_engine/startosis_packages/mock_package_content_provider/mock_package_content_provider.go (hunk 3 above)
Comment (h4ck3rk3y) [reply core/server/api_container/server/startosis_engine/startosis_packages/mock_package_content_provider/mock_package_content_provider.go]: Oh good find. Will remove.

Comment (h4ck3rk3y) [reply core/server/api_container/server/startosis_engine/startosis_packages/mock_package_content_provider/mock_package_content_provider.go]: Oh it needs to be made to work with relative paths. Added todo.

**Pull request merged**